src/engraving/rendering/dev/beamtremololayout.cpp

```diff
@@ -813,72 +888,64 @@ bool BeamTremoloLayout::calculateAnchorsCross(const BeamBase* item, BeamBase::La
-                // if outside notes on top or bottom staff are on the same staff line, slope = 0
-                // no further adjustment needed, the beam is already well-placed and horizontal
-                forceHoriz = true;
-            }
-            // otherwise, we have to compare the slopes from the top staff and bottom staff.
+
             int topSlant = topFirstLine - topLastLine;
             if (constrainTopToQuarter && topSlant != 0) {
                 topSlant = topFirstLine < topLastLine ? -1 : 1;
             }
+            if (!topLast) {
+                // If there's only one note, set direction to neutral
+                topSlant = 0;
+            }
             int bottomSlant = bottomFirstLine - bottomLastLine;
             if (constrainBottomToQuarter && bottomSlant != 0) {
                 bottomSlant = bottomFirstLine < bottomLastLine ? -1 : 1;
             }
+            if (!bottomLast) {
+                // If there's only one note, set direction to neutral
+                bottomSlant = 0;
+            }
             if ((maxMiddleTopLine >= std::max(topFirstLine, topLastLine)
                  || (minMiddleBottomLine <= std::min(bottomFirstLine, bottomLastLine)))) {
                 forceHoriz = true;
             }
-            if (topSlant == 0 || bottomSlant == 0 || forceHoriz) {
-                // if one of the slants is 0, the whole slant is zero
-            } else if ((topSlant < 0 && bottomSlant < 0) || (topSlant > 0 && bottomSlant > 0)) {
-                int slant = (std::abs(topSlant) < std::abs(bottomSlant)) ? topSlant : bottomSlant;
-                slant = std::min(std::abs(slant), getMaxSlope(ldata));
-                double slope = slant * ((topSlant < 0) ? -quarterSpace : quarterSpace);
+
+            const int topSlantDir = topSlant <= 0 ? (topSlant == 0 ? 0 : -1) : 1;
+            const int bottomSlantDir = bottomSlant <= 0 ? (bottomSlant == 0 ? 0 : -1) : 1;
+            // Check all directions which aren't neutral (0) match
+            // If there is any contradiction between top staff direction, bottom direction, beam side switch direction and staff switch direction, force horizontal
+            // If all are neutral, force horizontal
+            int overallDirection = 0;
+            for (int direction : { topSlantDir, bottomSlantDir, beamSideSwitchDirection, staffSwitchDirection }) {
+                if (direction != 0) {
+                    overallDirection = direction;
+                    break;
+                }
+            }
+            forceHoriz = forceHoriz || overallDirection == 0
+                         || ((topSlantDir != 0 && topSlantDir != overallDirection)
+                             || (bottomSlantDir != 0 && bottomSlantDir != overallDirection)
+                             || (beamSideSwitchDirection != 0 && beamSideSwitchDirection != overallDirection)
+                             || (staffSwitchDirection != 0 && staffSwitchDirection != overallDirection));
```

Comment:
Try to avoid these big lists of conditions if you can. Sometimes breaking these down into intermediate bool variables helps make it more readable and simpler to debug